cinema/views.py

```diff
@@ -20,25 +22,52 @@
     OrderListSerializer,
 )
 
+from cinema.permissions import IsAdminOrIfAuthenticatedOrReadOnly
 
-class GenreViewSet(viewsets.ModelViewSet):
+
+class GenreViewSet(
+    viewsets.GenericViewSet,
+    mixins.CreateModelMixin,
+    mixins.ListModelMixin,
+):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication, )
+    permission_classes = (IsAdminOrIfAuthenticatedOrReadOnly, )
```

Comment:
please use DEFAULT_PERMISSION_CLASS and DEFAULT_AUTHENTICATION_CLASSES in settings to avoid duplication in all classes